proposals/flexible-vectors/FlexibleVectors.md

```diff
@@ -0,0 +1,327 @@
+Flexible vectors overview
+=========================
+
+The goal of this proposal is to provide flexible vector instructions for
+WebAssembly as a way to bridge the gap between existing SIMD instruction sets
+available on various platforms. `simd128` proposal already identified
+operations that would commonly work on platforms that are important to
+WebAssembly, this proposal is attempting to extend the same operations to work
+with variable vector lengths.
+
+## Types
+
+Proposal introduces the following vector types:
+
+- `vec.v8` : 8-bit lanes
+- `vec.v16`: 16-bit lanes
+- `vec.v32`: 32-bit lanes
+- `vec.v64`: 64-bit lanes
+
+### Lane division interpretation
+
+In semantic pseudocode `S` is the particular vector type, `S.LaneBits` is the
+size of the lane in bits, `S.Lanes` is the number of lanes, which is dynamic.
```

Comment:
nit. Wondering if there are any tradeoffs for S.LaneBytes vs. S.LaneBits. :)

Reply:
This comes from imaginary Python library that we "use" when writing pseudocode for the operations (just like in `simd128`), so either or even both would work 😄